src/server/scripts/EasternKingdoms/zone_searing_gorge.cpp

```diff
@@ -0,0 +1,143 @@
+{
+public:
+    go_dark_iron_pillow_event_generator() : GameObjectScript("go_dark_iron_pillow_event_generator") {}
+
+
+    struct go_dark_iron_pillow_event_generatorAI : public GameObjectAI
+    {
+        go_dark_iron_pillow_event_generatorAI(GameObject* go) : GameObjectAI(go), owner(go) {};
+
+        void InitializeAI() override
+        {
+            events.ScheduleEvent(EVENT_WAVE_1,      60000, 0, 0);
+            events.ScheduleEvent(EVENT_WAVE_2,     180000, 0, 0);
+            events.ScheduleEvent(EVENT_CHAMBERMAID,300000, 0, 0);
+        }
+        void UpdateAI(uint32 diff) override
+        {
+            events.Update(diff);
+            while(uint32 eventId = events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_WAVE_1:
+                    for (size_t i = 0; i < 10; i++)
+                    {
+                        if (TempSummon* summon = owner->SummonCreature(NPC_SLEEPY_DARK_IRON_WORKER, -6568.03f, -1306.73f, 208.70f, 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 120000))
+                        {
+                            summon->ToCreature()->GetMotionMaster()->MovePath(NPC_SLEEPY_DARK_IRON_WORKER*100+i, true);
+                            if (i == 0)
+                                summon->ToCreature()->AI()->Talk(0);
+                        }
+                    }
+                    break;
+                case EVENT_WAVE_2:
+                    for (size_t i = 0; i < 10; i++)
+                    {
+                        if (TempSummon* summon = owner->SummonCreature(NPC_SLEEPY_DARK_IRON_WORKER, -6481.22f, -1349.06f, 212.10f, 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 120000))
+                        {
+                            summon->ToCreature()->GetMotionMaster()->MovePath(NPC_SLEEPY_DARK_IRON_WORKER * 100+10+i, true);
+
+                            if (i == 0)
+                                summon->ToCreature()->AI()->Talk(1);
+                        }
+                    }
+                    break;
+                case EVENT_CHAMBERMAID:
+                    TempSummon* summon = owner->SummonCreature(NPC_CHAMBERMAID_PILLACLENCHER, -6545.88f, -1345.29f, 208.89f,0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 300000);
+                    break;
+                }
+            }
+        }
+
+        EventMap events;
+        GameObject *owner;
+    };
+
+    GameObjectAI* GetAI(GameObject* go) const override
+    {
+        return new go_dark_iron_pillow_event_generatorAI(go);
+    }
+};
+
+
+static std::mutex triggerLock;
+class go_dark_iron_pillow : public GameObjectScript
+{
+public:
+    go_dark_iron_pillow() : GameObjectScript("go_dark_iron_pillow") {}
+
+
+    struct go_dark_iron_pillowAI : public GameObjectAI
+    {
+        go_dark_iron_pillowAI(GameObject* go) : GameObjectAI(go) {};
+
+        bool GossipHello(Player* player) override
+        {
+            std::lock_guard<std::mutex> triggerGuard(triggerLock);
```

Comment:
Do I need this lock?
What's the threading model for scripts, what happens if 2 players click a different pillow simultaneously?

Reply:
I don't know, but maybe some of our fellow users could test this for you.
I would have tested that if it wasn't for the fact that I play alone on my PC.
To me, this PR looks like long-missed good fun in the code. :)